pkg/util/pflagaliases.go

```diff
@@ -0,0 +1,64 @@
+package util
+
+import "github.com/spf13/pflag"
+
+func AddFlagAliasesString(flags *pflag.FlagSet, originalFlag string, aliasMap map[string][]string, aliases ...string) {
+	f := flags.Lookup(originalFlag)
+	if f == nil {
+		panic("bug! AddFlagAliasesString couldn't find original flag in collection")
+	}
+	for _, alias := range aliases {
+		flags.String(alias, f.DefValue, "")
+		_ = flags.MarkHidden(alias)
+	}
+	aliasMap[originalFlag] = aliases
+}
+
+func AddFlagAliasesBool(flags *pflag.FlagSet, originalFlag string, aliasMap map[string][]string, aliases ...string) {
+	f := flags.Lookup(originalFlag)
+	if f == nil {
+		panic("bug! AddFlagAliasesBool couldn't find original flag in collection")
+	}
+	for _, alias := range aliases {
+		flags.Bool(alias, false, "") // this would be broken if we had any bools with default value of true, but we don't
+		_ = flags.MarkHidden(alias)
+	}
+	aliasMap[originalFlag] = aliases
+}
+
+func AddFlagAliasesStringSlice(flags *pflag.FlagSet, originalFlag string, aliasMap map[string][]string, aliases ...string) {
+	f := flags.Lookup(originalFlag)
+	if f == nil {
+		panic("bug! AddFlagAliasesStringSlice couldn't find original flag in collection")
+	}
+	for _, alias := range aliases {
+		flags.StringSlice(alias, nil, "")
+		_ = flags.MarkHidden(alias)
+	}
+	aliasMap[originalFlag] = aliases
+}
+
+func ApplyFlagAliases(flags *pflag.FlagSet, aliases map[string][]string) {
+	// find values that may have been specified using a flag alias, and copy the values across to the primary flags
+	for k, v := range aliases {
+		primaryFlag := flags.Lookup(k)
+		for _, aliasName := range v {
+			aliasFlag := flags.Lookup(aliasName)
+			aliasValueString := aliasFlag.Value.String() // flags get stringified here, but it's fast enough and a one-shot so meh
+			if aliasValueString != aliasFlag.DefValue {
+				// we have to call set because .Value holds the pointer to the bound variable;
+				// if we set one Value to another we end up pointing at different storage and it doesn't work
+
+				if aliasFlag.DefValue == "[]" && len(aliasValueString) > 2 && aliasValueString[0] == '[' {
```

Comment:
This code is 🤮 and will break instantly, should survey change the way they print flags internally, or anything like that. But it's the best we can do with the API surface that survey gives to us; Unit tests will catch it if something ever does break so 🤷